mmcv/ops/csrc/common/cuda/stack_ball_query_cuda_kernel.cuh

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) OpenMMLab. All rights reserved
+// Modified from
+// https://github.com/sshaoshuai/Pointnet2.PyTorch/tree/master/pointnet2/src/ball_query_gpu.cu
+#ifndef STACK_BALL_QUERY_CUDA_KERNEL_CUH
+#define STACK_BALL_QUERY_CUDA_KERNEL_CUH
+
+#ifdef MMCV_USE_PARROTS
+#include "parrots_cuda_helper.hpp"
+#else
+#include "pytorch_cuda_helper.hpp"
+#endif
+
+__global__ void stack_ball_query_forward_cuda_kernel(int B, int M, float radius, int nsample,
+                                  const float *new_xyz,
+                                  const int *new_xyz_batch_cnt,
+                                  const float *xyz, const int *xyz_batch_cnt,
+                                  int *idx) {
+  // :param xyz: (N1 + N2 ..., 3) xyz coordinates of the features
+  // :param xyz_batch_cnt: (batch_size), [N1, N2, ...]
+  // :param new_xyz: (M1 + M2 ..., 3) centers of the ball query
+  // :param new_xyz_batch_cnt: (batch_size), [M1, M2, ...]
+  // output:
+  //      idx: (M, nsample)
+  int pt_idx = blockIdx.x * blockDim.x + threadIdx.x;
+  if (pt_idx >= M)
+    return;
+
+  int bs_idx = 0, pt_cnt = new_xyz_batch_cnt[0];
+  for (int k = 1; k < B; k++) {
+    if (pt_idx < pt_cnt)
```

Comment:
Errr, is it possible that `new_xyz_batch_cnt[k]<0` ? If not, we should move this branch outside the loop.

Reply:
`new_xyz_cnt` is num of points, so it is always >= 0.